### PR TITLE
IDEMPIERE-4520: Attribute Instance not Refreshed on Create

### PR DESCRIPTION
**Pull request by dpansheriya**
This is bug fix reported by Norbert.

### CODE REVIEW

File: org.adempiere.base/src/org/compiere/model/MAttribute.java [hunk 1]
```diff
@@ -280,7 +280,12 @@ public void setMAttributeInstance(int M_AttributeSetInstance_ID, int valueInt, S
 	{
 		MAttributeInstance instance = getMAttributeInstance(M_AttributeSetInstance_ID);
 		if (instance == null)
+		{
```
Comment (hengsin) [inline org.adempiere.base/src/org/compiere/model/MAttribute.java:283]: Going through the history, this originated from the changes for IDEMPIERE-2999. While the code works fine (including this fix), it is not tidy design wise. 1. You really want to take a KeyNamePair, not int valueInt + String value parameter that's hard to understand. 2. MAttributeInstance have getValueInt and setValueInt pair that doesn't follows common Java Bean rule and kind of confusing. 3. MAttributeInstance should have a constructor that take KeyNamePair (or valueInt + value for the current implementation). The lack of that constructor make it inconsistent with other part of the code.

**Comment by CarlosRuiz-globalqss** [comment]
Replaced with https://github.com/idempiere/idempiere/pull/409

I reopened the ticket [IDEMPIERE-2999](https://idempiere.atlassian.net/browse/IDEMPIERE-2999) and added the comments from @hengsin 

@dpansheriya can you please check the recommendations there?

**Comment by dpansheriya** [comment]
> Replaced with #409
> 
> I reopened the ticket [IDEMPIERE-2999](https://idempiere.atlassian.net/browse/IDEMPIERE-2999) and added the comments from @hengsin
> 
> @dpansheriya can you please check the recommendations there?

Should this suggestions committed about IDEMPIERE-2999?

**Comment by CarlosRuiz-globalqss** [comment]
Yes, I think is better as is part of that ticket

File: org.adempiere.base/src/org/compiere/model/MAttribute.java (hunk 1 above)
Comment (dpansheriya) [reply org.adempiere.base/src/org/compiere/model/MAttribute.java:283]: 1. We follow syntas matching to public void setMAttributeInstance (int M_AttributeSetInstance_ID, MAttributeValue value) which was existing method. 
Do you here suggest we should replace all setMAttributeInstance by single one accepting KeyNamePair?
2. We are using same column named ValueNumber for storing Numeric, Integer or Foreignkey reference to reduce number of columns. This cause as to create method for Int but there is no column of Integer presents. Also we are passing string value along with Integer value for storing readable attribute descriptions to the attribute instance for Table Direct or Table type references.
3. Yes, I think we should do this. Will take care for this.

**Pull request closed without merging**